core/services/gateway/network/httpclient.go

```diff
@@ -0,0 +1,105 @@
+package network
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+// HttpClient interfaces defines a method to send HTTP requests
+// handles retries and timeouts
+type HttpClient interface {
+	Send(ctx context.Context, req HttpRequest) (*HttpResponse, error)
+}
+
+type HttpRequest struct {
+	Method     string
+	URL        string
+	Headers    map[string]string
+	Body       []byte
+	Timeout    time.Duration
+	RetryCount uint8
+}
+
+type HttpResponse struct {
+	StatusCode int               // HTTP status code
+	Headers    map[string]string // HTTP headers
+	Body       []byte            // Base64-encoded binary body
```

Comment:
It's not rally base64-encoded here yet, since it's a byte array not a string.

Reply:
oops copy-pasta. fixed.